src/core/background/action.ts

```diff
@@ -103,20 +101,45 @@ function getIconURL(url: string) {
  * @param mode - Controller mode to set state to display
  * @param song - Currently playing song, if there is one
  */
-function setAction(mode: ControllerModeStr, song: CloneableSong | null): void {
+async function setAction(tab: ManagerTab): Promise<void> {
 	let songstr = '';
-	if (song) {
-		const clonedSong = new ClonedSong(song, -1);
+	if (tab.song) {
+		const clonedSong = new ClonedSong(tab.song, -1);
 		songstr = `${clonedSong.getArtist()} - ${clonedSong.getTrack()}`;
 	}
 
 	browser.action.setIcon({
 		path: {
-			19: getIconURL(`icons/page_action_${mode.toLowerCase()}_19.png`),
-			38: getIconURL(`icons/page_action_${mode.toLowerCase()}_38.png`),
+			16: getIconURL(
+				`icons/action_${tab.mode.toLowerCase()}_16_${await getIconType()}.png`
```

Comment:
Should we add like

`const iconType = await getIconType();` 

And reuse that to avoid having to call `getIconType` over and over?

Reply:
Done, also made some other changes to the same part in the spirit of DRY